File: ipo.cc
// IPO
#define FOR(i, a, b) for (remove_cv<remove_reference<decltype(b)>::type>::type i = (a); i < (b); i++)
#define REP(i, n) FOR(i, 0, n)

class Solution {
public:
  int findMaximizedCapital(int k, int W, vector<int>& Profits, vector<int>& Capital) {
    vector<pair<int, int>> a;
    REP(i, Profits.size())
      a.emplace_back(Capital[i], Profits[i]);
    sort(a.begin(), a.end());
 

File: ipo.rs
// IPO
impl Solution {
    pub fn find_maximized_capital(k: i32, mut w: i32, profits: Vec<i32>, capital: Vec<i32>) -> i32 {
        use std::collections::BinaryHeap;
        let mut h = BinaryHeap::new();
        let mut a: Vec<(i32,i32)> = (0..profits.len()).map(|i| (capital[i], profits[i])).collect();
        a.sort_unstable();
        let mut i = 0;
     